### PR TITLE
Pwm doc

### PR DESCRIPTION
**Pull request by rasmus-madsen**
this pull request has all the documentation updates for V1
additionally, it has a few code updates to fix some things I found while writing the documentation.

this review will serve as the PWM V1 review

### CODE REVIEW

File: hw/ip/pwm/data/pwm_testplan.hjson [hunk 1]
```diff
@@ -2,111 +2,224 @@
...
 
             Stimulus:
-              - Program pwm enable register (PWM_EN)
-              - Program configuration registers (CFG, INVERT)
-              - Disable blink and heartbeat enable registers (these mode are verified with other tests)
-              - Program REGEN to lock (enable) register settings and start pwm function
+            - run segments on one channel in standard mode
+            - both send read and write commands
+            - reads and writes the Host FIFO's
+                - uses different clock divisions and resolutions
```
Comment (martin-lueker) [inline hw/ip/pwm/data/pwm_testplan.hjson:21]: Did this get accidentally added from SPI_HOST?  What are these fifos?  Read and write commands? Segments?  I think the previous stimulus makes much more sense.

**Review by martin-lueker — COMMENTED** [review]
Mostly looks good to me except for my first comment about the SPI-ish description of the first stimulus.

File: hw/ip/pwm/data/pwm_testplan.hjson (hunk 1 above)
Comment (rasmus-madsen) [reply hw/ip/pwm/data/pwm_testplan.hjson:21]: I have no clue what happended - I was task switching between the do a few days back - must have accidentally thought I was doing spi but still that would also not make sense. I have updated it to what it should be

**Review by martin-lueker — APPROVED** [review]
Great.  Thanks Rasmus.  LGTM

**Pull request merged**